nomos-services/data-availability/sampling/src/lib.rs

```diff
@@ -182,10 +193,15 @@ where
                 tokio::select! {
                     Some(service_message) = service_state.inbound_relay.recv() => {
                         Self::handle_service_message(service_message, &mut network_adapter, &mut sampler).await;
+                        // cleanup not on time samples
+                        sampler.prune();
                     }
                     Some(sampling_message) = sampling_message_stream.next() => {
                         Self::handle_sampling_message(sampling_message, &mut sampler).await;
+                        // cleanup not on time samples
+                        sampler.prune();
```

Comment:
As I understand, pruning now is done only after some interaction with the service. Could there be a situation in which the sampling was triggered way before the `old_blobs_check_duration`, for some reason no other interactions happen with the service, and block producer requests for outdated validated blob (because pruning is done after the message is handled). Maybe we should have a ticker for pruning?

Reply:
A ticker is a good idea probably. Let's do this, lets delegate ticker creation to the backend (we need to add the method to the trait), because is the one that should know about timings, then in the service on tick we prune. Nice catch @bacv !

Reply:
Yes that was the initial idea with the thread in the backend, which we then moved out to the service. A combination of the two sounds like the best approach. If I am understanding correctly, a ticker in the backend, but executing pruning from the service.

Reply:
The backend is just responsible of building the ticker (as it holds the proper configuration), and pruning. The service will call the prune when the ticker ticks, as it owns the main loop.